### ISSUE TITLE
Does this support TransactionFlow?

### ISSUE THREAD

**Issue by ramuambati** [CoreWCF/CoreWCF#29]
Hi,

I do not see TransactionFlow attribute in NetTcpBinding. Is this is expected? I see same in .NET Framework. If this is correct, how do I use Transactions from .net standard to call WCF Service? Thanks

             var tcpb = new NetTcpBinding();
                    tcpb.**_TransactionFlow_** = true;

**Comment by mconnew** [comment]
@ramuambati , if you are still interested in this, can you add your vote here: https://github.com/CoreWCF/CoreWCF/issues/234#issuecomment-703430017

**Comment by MilanRaval** [comment]
@mconnew could you check below implementation, do you think there is any issue in the implementation ? Sometimes CoreWCF call never returns to the client and just times out.

OperationBehavior Attribute 
 ```
[AttributeUsage(AttributeTargets.Method)]
 public sealed class OperationBehaviorAttribute : Attribute, IOperationBehavior
 {
     public bool TransactionScopeRequired { get; set; } = true;

     public void AddBindingParameters(OperationDescription operationDescription, BindingParameterCollection bindingParameters)
     {
         // ignore
     }

     public void ApplyClientBehavior(OperationDescription operationDescription, ClientOperation clientOperation)
     {
         // ignore
     }

     public void ApplyDispatchBehavior(OperationDescription operationDescription, DispatchOperation dispatchOperation)
     {
         if (TransactionScopeRequired)
         {
             dispatchOperation.Invoker = new TransactionScopeOperationInvoker(dispatchOperation.Invoker);
         }
     }

     public void Validate(OperationDescription operationDescription)
     {
         // ignore
     }
 }
```

TransactionScopeOperationInvoker - above attribute uses it to invoke operation under a transaction scope. Assuming here that client is not starting any ambient transaction
```
 public class TransactionScopeOperationInvoker : IOperationInvoker
 {
     private readonly IOperationInvoker _baseInvoker;

     public TransactionScopeOperationInvoker(IOperationInvoker baseInvoker)
     {
         _baseInvoker = baseInvoker;
     }

     public object[] AllocateInputs()
     {
         return _baseInvoker.AllocateInputs();
     }

     public async ValueTask<(object returnValue, object[] outputs)> InvokeAsync(object instance, object[] inputs)
     {
         try
         {
             using (var scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
             {
                 var (returnValue, outputs) = await _baseInvoker.InvokeAsync(instance, inputs);
                 scope.Complete();
                 return (returnValue, outputs);
             }
         }
         catch (System.Exception ex)
         {
             System.Diagnostics.Debug.WriteLine(ex.Message);
             throw;
         }
     }
 }
```
Attribute usage on a method 
```
 [OperationBehavior(TransactionScopeRequired = true)]
 public Customer TestMethod(Customer c) {
   // code to update customer value in the database and few other operations
 } 
```

**Comment by BryanAldrich** [comment]
> Unfortunately System.Transactions in .NET Core doesn't support distributed transactions. It only supports using a transaction scope when using a SQL client (might even be limited to SQL Server, not quite sure how broad the SQL support is) to roll back multiple SQL operations. I would happily add the support (it's actually not a lot of code in WCF to use transactions) if/when support is added. The issue in .NET Core tracking this is dotnet/corefx#13532. Add your support over there. I will say that it's highly unlikely distributed transactions will be supported on anything other than Windows. But you never know, somebody might make it work for Linux.

I hate to necro an old thread,  but .net core has windows only msdtc transactions since v7. Is wcf flow support still an option?

**Comment by mconnew** [comment]
I've got a bit of news for you. I've been experimenting with using AI to help with porting implementation across from .NET Framework. Your question reminded me this is on the long backlog. To be able to add this to CoreWCF, we need to be able to have tests, which means we need client support. I just ported transactions support for the WCF Client as can be seen in this PR: https://github.com/dotnet/wcf/pull/5926  
Once that's completed and we have packages available, I can do the same for CoreWCF. It's not going to be an immediate thing, so don't expect next week, but with how far AI has come recently, it's becoming a lot more feasible to port more features and I'm less of a bottleneck.
